Source/MissionSystem/Private/MSMissionSystem.cpp

```diff
@@ -173,6 +206,29 @@ void UMSMissionSystem::WhenMissionEnds( UMSMissionData * mission_data, const FMS
     MissionEndObservers.Emplace( MoveTemp( observer ) );
 }
 
+void UMSMissionSystem::WhenMissionObjectiveStartsOrIsActive( TSubclassOf< UMSMissionObjective > mission_objective, const FMSMissionSystemMissionObjectiveStartsDelegate & when_mission_objective_starts )
+{
+    FMissionObjectiveStartObserver observer;
+    observer.MissionObjective = mission_objective;
+    observer.Callback = when_mission_objective_starts;
+
+    MissionObjectiveStartObservers.Emplace( MoveTemp( observer ) );
+
+    if ( IsMissionObjectiveActive( mission_objective ) )

```

Comment:
I would place this at the top and do an early return.
We don't need to add the observer if the mission objective is already active and we executed it already.

Reply:
Done